### DISCUSSION TITLE
Example of MongoCodecProvider[MyClass]

### DISCUSSION THREAD

**Discussion by sc6l6d3v** [Kirill5k/mongo4cats#18] [Q&A] [answered]
Hi,
Both the distinct as well as aggregate pages on documentation make mention of a 

// assuming you have an instance of MongoCodecProvider[MyClass] available in the implicit scope

As it is easier to contrive a distinct example, could you share a test or perhaps a small code fragment showing what this CodeProvider.get() method should be doing?

Thanks,
Henry

**Comment by Kirill5k** [comment, marked as answer]
Hello.

Effectively, `MongoCodecProvider[T]` is just a wrapper around MongoDB's native `CodecProvider` class. The only reason why I've introduced it is so that I could derive codecs for case classes that have circe codecs provided (i.e. [here](https://github.com/Kirill5k/mongo4cats/blob/master/circe/src/main/scala/mongo4cats/circe/MongoJsonCodecs.scala#L51)).

An example of how it is done can be found [here](https://github.com/Kirill5k/mongo4cats/blob/master/examples/src/main/scala/mongo4cats/examples/CaseClassesWithCirceCodecs.scala).
There is also a page written about it [here](https://kirill5k.github.io/mongo4cats/docs/circe.html).

**Reply by sc6l6d3v** [reply]
Thanks for the reply. Will adopt your example.
